sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,921 @@
+  private ruleResourceSerializer: AtomXmlSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param connectionString The connection string needed for the client to connect to Azure.
+   * @param options ServiceBusAtomManagementClientOptions
+   */
+  constructor(connectionString: string, options?: ServiceBusAtomManagementClientOptions) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+
+    if (connectionStringObj.Endpoint == undefined) {
+      throw new Error("Endpoint must be supplied in the connection string");
+    }
+
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(log.httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (options && options.proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(options.proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
+  }
+
+  /**
+   * Creates a queue with given name, configured using the given options
+   * @param queueName
+   * @param queueOptions Options to configure the Queue being created.
+   * For example, you can configure a queue to support partitions or sessions
+   */
+  async createQueue(queueName: string, queueOptions?: QueueOptions): Promise<CreateQueueResponse> {
+    const response: HttpOperationResponse = await this._putResource(
+      queueName,
+      buildQueueOptions(queueOptions || {}),
+      this.queueResourceSerializer,
+      false
+    );
+    log.httpAtomXml(
+      `Performing management operation - createQueue() for "${queueName}" with options: ${queueOptions}`
+    );
+    return this.buildQueueResponse(response);
+  }
+
+  /**
+   * Returns an object representing the Queue with the given name along with all its properties
+   * @param queueName
+   */
+  async getQueue(queueName: string): Promise<GetQueueResponse> {
+    const response: HttpOperationResponse = await this._getResource(
+      queueName,
+      this.queueResourceSerializer
+    );
+    log.httpAtomXml(`Performing management operation - getQueue() for "${queueName}"`);
```

Comment:
I would move the log operations for all of these before the service calls. At this point the management operation has already been completed, and if it errors we'll never trigger this log.

Reply:
addressed